File: webservices/src/main/java/fieg/modulos/administrativo/pagamentosnaobaixados/dto/PagamentoNaoBaixadoDTO.java
package fieg.modulos.administrativo.pagamentosnaobaixados.dto;

import fieg.modulos.formapagamento.enums.FormaPagamentoSimplificado;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class PagamentoNaoBaixadoDTO {
    private FormaPagamentoSimplificado formaPagamento;
    private LocalDateTime dataPagamento;
    private String sacadoCpfCnpj;
    private String sacadoNome;
    private Integer idUnidade;
    private BigDecimal valorOperacao;
    private Integer parcelamento;
    private Integer idOperadorRecebimento;
    private Integer idPedido;

    private List<InfoCobrancaPagamentoNaoBaixadoDTO> cobrancas;
}

File: web/src/models/DTOs/Outros/PagamentoNaoBaixadoDTO.ts
import { FormaPagamentoSimplificado } from "./FormasDePagamentoSimplificas";
import { UnidadeDTO } from "./UnidadeDTO";

export interface PagamentoNaoBaixadoComUnidadeDTO {
  pagamentoNaoBaixado: PagamentoNaoBaixadoDTO;
  unidade: UnidadeDTO;
}

export interface PagamentoNaoBaixadoDTO {
  formaPagamento: FormaPagamentoSimplificado;
  dataPagamento: Date;
  sacadoCpfCnpj: string;
  sacadoNome: string;
  idUnidade: number;
  valorOperacao: number;
  parcelamento: number;
  idOperadorRecebimento: number | null;
  idPedido: number | null;

  cobrancas: InfoCobrancaPagamentoNaoBaixadoDTO[];
}

export interface InfoCobrancaPagamentoNaoBaixadoDTO {
  idCobrancaCliente: number;
  contId: number;
  idSistema: number;
  numeroParcela: number;
  valorCobranca: number;
  situacaoParcela: string;
  statusContrato: string;
}
